decompiler/main.cpp

```diff
@@ -25,16 +26,55 @@ int main(int argc, char** argv) {
   file_util::init_crc();
   init_opcode_info();
 
-  if (argc != 4) {
-    printf("Usage: decompiler <config_file> <in_folder> <out_folder>\n");
+  if (argc < 4) {
+    printf(
+        "Usage: decompiler <config_file> <in_folder> <out_folder> "
+        "[bool_flag_name=true/false...]\n");
     return 1;
   }
   fmt::print("[Mem] After init: {} MB\n", get_peak_rss() / (1024 * 1024));
 
   // collect all files to process
   Config config;
   try {
-    config = read_config_file(argv[1]);
+    // Allow overriding config boolean flags via CLI
+    // There are very minimum guard-rails here
+    //
+    // "<key>=<override>"
+    //
+    // This allows us to run scripts that deviate from the defaults
+    std::map<std::string, bool> overrides;
+    if (argc > 4) {
+      for (int i = 4; i < argc; i++) {
+        std::string val = argv[i];
+        if (val.find('=') == std::string::npos) {
+          printf("Aborting - invalid flag override syntax\n");
+          printf(
+              "Usage: decompiler <config_file> <in_folder> <out_folder> "
+              "[bool_flag_name=true/false...]\n");
+          return 1;
+        }
+        auto pair = split_string(argv[i], '=');
+        if (pair.size() > 2) {
+          printf("Aborting - invalid flag override syntax, provide pairs!\n");
+          printf(
+              "Usage: decompiler <config_file> <in_folder> <out_folder> "
+              "[bool_flag_name=true/false...]\n");
+          return 1;
+        }
+        if (pair.at(1) != "true" && pair.at(1) != "false") {
+          printf("Aborting - invalid flag override syntax, true|false only!\n");
+          printf(
+              "Usage: decompiler <config_file> <in_folder> <out_folder> "
+              "[bool_flag_name=true/false...]\n");
+          return 1;
+        }
+        overrides.insert({pair.at(0), pair.at(0) == "true"});
+      }
+    }
```

Comment:
this is fine for now, but if we want to make this more complicated it might be a good idea to bring in an argument parsing library.